serrors/serrors.go

```diff
@@ -17,38 +23,68 @@ func (e *Error) Unwrap() error {
 
 // Error returns the string representation of the error
 func (e *Error) Error() string {
-	return e.error.Error()
+	var elems []string
+	keys := lo.Keys(e.keysAndValues)
+	sort.Strings(keys) // sort keys so we always get a consistent ordering
+	for _, k := range keys {
+		v := e.keysAndValues[k]
+		elems = append(elems, fmt.Sprintf("%s=%v", k, v))
+	}
+	return fmt.Sprintf("%s (%s)", e.error.Error(), strings.Join(elems, ", "))
 }
 
 // WithValues injects additional structured keys and values into the error
 func (e *Error) WithValues(keysAndValues ...any) *Error {
-	e.keysAndValues = append(e.keysAndValues, keysAndValues...)
+	for i := 0; i < len(keysAndValues); i += 2 {
```

Comment:
It's not specific to this PR, but while we're here should we add a panic if an odd number of elements are passed, same as in `Wrap`?

Reply:
Good callout -- added the validation